hudi-common/src/main/java/org/apache/hudi/common/util/OrcUtils.java

```diff
@@ -100,6 +126,9 @@
           String rowKey = rowKeys.toString(i);
           String partitionPath = partitionPaths.toString(i);
           hoodieKeys.add(new HoodieKey(rowKey, partitionPath));
+          if (batchSize > 0 && hoodieKeys.size() >= batchSize) {
+            break;
+          }
```

Comment:
1. I see that there is already a batchSize param in `VectorizedRowBatch` and the default val is 1024, can we use that
2. We should not skip the for-loop early in while reading a batch, which would cause data lost, instead we can put it out of the for-loop, that means we must read a full `VectorizedRowBatch`
3. Considering there is already a batch there, the more reasonable way to control the read buffer is a param like `int batches`, which means the number of read batches once a time. Or you can divide the `batchSize` by the `VectorizedRowBatch` size to get the vector batch number to read.